appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Shadow.kt

```diff
@@ -0,0 +1,55 @@
+package com.bumble.appyx.interactions.core.ui.property.impl
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.AnimationVector1D
+import androidx.compose.animation.core.Easing
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.unit.dp
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.math.lerpFloat
+import com.bumble.appyx.interactions.core.ui.property.Interpolatable
+import com.bumble.appyx.interactions.core.ui.property.MotionProperty
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class Shadow(
+    uiContext: UiContext,
+    target: Target,
+    visibilityThreshold: Float = 0.01f,
+    displacement: StateFlow<Float> = MutableStateFlow(0f),
+) : MotionProperty<Float, AnimationVector1D>(
+    uiContext = uiContext,
+    animatable = Animatable(target.value),
+    easing = target.easing,
+    visibilityThreshold = visibilityThreshold,
+    displacement = displacement
+), Interpolatable<Shadow.Target> {
+
+    class Target(
+        val value: Float,
+        val easing: Easing? = null,
+    ) : MotionProperty.Target
+
+    override fun calculateRenderValue(base: Float, displacement: Float): Float =
+        base - displacement
+
+    override val visibilityMapper: ((Float) -> Boolean) = { true }
```

Comment:
Does shadow value equals to 0 means that there is no shadow present?

Reply:
Yes, that is correct